Add test for archive object replacement

francoisferrand opened the pull request:
- Add test for archive object replacement
- Bump cloudserver

Issue: ZENKO-4928

francoisferrand committed:
Add artesca-root-ca-issuer cluster issuer

Issue: ZENKO-4876

francoisferrand committed:
Test that federated metrics are retrieved

Issue: ZENKO-4876

francoisferrand committed:
Set federated prometheus name

This is needed to avoid conflict with internal prometheus service,
since we are running in the same cluster.

Issue: ZENKO-4876

francoisferrand committed:
Fix prometheus config

ServiceMonitor were not scrapped.

Issue: ZENKO-4876

francoisferrand committed:
Fix zkop install to support sha1

When an SHA1 is used which is not the tip of the branch, it is not
available after a `git clone` (which fetches only commits reachable
from current branches).

This causes issues if a build is started while the component branch with
the commit has been rewritten (rebase...)

To overcome this, the safe way is to explicitely fetch the commit: which
works for any commit (branch, tag, sha1).

Issue: ZENKO-4876

francoisferrand committed:
Merge alerts workflow

No reason to keep separate, and it helps avoid a github bug/limitation
when creating a check: and the

Issue: ZENKO-4876

francoisferrand committed:
Move kafka dump to folder

Issue: ZENKO-4876

francoisferrand committed:
Bump backbeat 8.6.51

Issue: ZENKO-4926

francoisferrand committed:
Bump zenko-operator 1.6.4

Issue: ZENKO-4926

francoisferrand committed:
Add test for archive object replacement

Add replacement with MPU or copyObject.

Issue: ZENKO-4928

Seragonia committed:
Ensure callback is called once in doWhilst with S3 operations

The trigger seem to be
https://github.com/aws/aws-sdk-js/issues/1678
... causing the callback to be called twice, which
is a problem in these loops.
Issue: ZENKO-4925

Seragonia committed:
Improve variable in doWhilst

Issue: ZENKO-4925

bert-e commented:
# Hello francoisferrand,



My role is to assist you with the merge of this
pull request. Please type `@bert-e help` to get information
on this process, or consult the [user documentation](
https://bert-e.scality.net/scality/zenko/doc/user).


<details>
  <summary><b>Available options</b></summary>

  name   | description  | privileged   | authored
  ------ | ------------ | ------------ |----------
  `/after_pull_request` | Wait for the given pull request id to be merged before continuing with the current one. |  | 
  `/bypass_author_approval` | Bypass the pull request author's approval |  :star:  | 
  `/bypass_build_status` | Bypass the build and test status |  :star:  | 
  `/bypass_commit_size` | Bypass the check on the size of the changeset ```TBA``` |  :star:  | 
  `/bypass_incompatible_branch` | Bypass the check on the source branch prefix |  :star:  | 
  `/bypass_jira_check` | Bypass the Jira issue check |  :star:  | 
  `/bypass_peer_approval` | Bypass the pull request peers' approval |  :star:  | 
  `/bypass_leader_approval` | Bypass the pull request leaders' approval |  :star:  | 
  `/approve` | Instruct Bert-E that the author has approved the pull request. |  |  :writing_hand: 
  `/create_pull_requests` | Allow the creation of integration pull requests. |  | 
  `/create_integration_branches` | Allow the creation of integration branches. |  | 
  `/no_octopus` | Prevent Wall-E from doing any octopus merge and use multiple consecutive merge instead |  | 
  `/unanimity` | Change review acceptance criteria from `one reviewer at least` to `all reviewers` |  | 
  `/wait` | Instruct Bert-E not to run until further notice. |  | 
  

</details>



<details>
  <summary><b>Available commands</b></summary>

  name   | description  | privileged
  ------ | ------------ | ------------
  `/help` | Print Bert-E's manual in the pull request. | 
  `/status` | Print Bert-E's current status in the pull request ```TBA``` | 
  `/clear` | Remove all comments from Bert-E from the history ```TBA``` | 
  `/retry` | Re-start a fresh build ```TBA``` | 
  `/build` | Re-start a fresh build ```TBA``` | 
  `/force_reset` | Delete integration branches & pull requests, and restart merge process from the beginning. | 
  `/reset` | Try to remove integration branches unless there are commits on them which do not appear on the source branch. | 
  
</details>




*Status report is not available.*

bert-e commented:
# Branches have diverged



This pull request's source branch `improvement/ZENKO-4928` has diverged from
`development/2.10` by more than 50 commits.

To avoid any integration risks, please re-synchronize them using one of the
following solutions:

* Merge `origin/development/2.10` into `improvement/ZENKO-4928`
* Rebase `improvement/ZENKO-4928` onto `origin/development/2.10`

Note: If you choose to rebase, you may have to ask me to rebuild
integration branches using the `reset` command.

williamlardier commented on .github/scripts/end2end/configs/zenkoversion.yaml:
```diff
@@ -9,9 +9,9 @@ spec:
-    cloudserver:
-      image: '${CLOUDSERVER_DASHBOARD}'
-      tag: '${CLOUDSERVER_TAG}'
+    # cloudserver:
+    #   image: '${CLOUDSERVER_DASHBOARD}'
+    #   tag: '${CLOUDSERVER_TAG}'
```
To be re-enabled before merging I guess

williamlardier commented on tests/ctst/features/dmf.feature:
```diff
@@ -92,3 +92,44 @@ Feature: DMF
+    @PreMerge
+    @Dmf
+    @ColdStorage
+    Scenario Outline: Overwriting of a cold mpu object
+    Given a "<versioningConfiguration>" bucket
+     And a transition workflow to "e2e-cold" location
+     And <objectCount> objects "obj" of size <objectSize> bytes
+    Then object "obj-1" should be "transitioned" and have the storage class "e2e-cold"
+     And dmf volume should contain <objectCount> objects
+    Given <objectCount> mpu objects "obj" of size <objectSize> bytes
+    Then object "obj-1" should be "transitioned" and have the storage class "e2e-cold"
+     And dmf volume should contain 1 objects
```
No need to indent `And` lines in gherkin

```suggestion
    Scenario Outline: Overwriting of a cold mpu object
    Given a "<versioningConfiguration>" bucket
    And a transition workflow to "e2e-cold" location
    And <objectCount> objects "obj" of size <objectSize> bytes
    Then object "obj-1" should be "transitioned" and have the storage class "e2e-cold"
    And dmf volume should contain <objectCount> objects
    Given <objectCount> mpu objects "obj" of size <objectSize> bytes
    Then object "obj-1" should be "transitioned" and have the storage class "e2e-cold"
    And dmf volume should contain 1 objects
```

williamlardier commented on tests/ctst/steps/utils/utils.ts:
```diff
@@ -199,6 +199,84 @@ async function createBucketWithConfiguration(
+    const result = await S3.copyObject(world.getCommandParameters());
+    const versionId = extractPropertyFromResults<string>(result, 'VersionId');
+    world.saveCreatedObject(key, versionId || '');
+    world.setResult(result);
+    return result;
```
Not strictly needed unless you check the result in a subsequent step; we usually use this setResult when we want, e.g., to check the authz result at the end, or get the expected return code, but I don't see such tests in your scenarios

williamlardier reviewed (approved):
Approving because the comments are not blocking & I know they will be handled before merging

KillianG commented on tests/ctst/steps/utils/utils.ts:
```diff
@@ -199,6 +199,83 @@ async function createBucketWithConfiguration(
     }
 }
 
+async function putMpuObject(world: Zenko, parts: number = 2, objectName: string, content?: string) {
```
Isn't there a minimum of size for parts of an MPU (IIRC its 5MB) ?. If so, should we log an error here if the test is misswritten or should we let S3 respond with an error ?

williamlardier replied:
We should let the test fail, actually the SDK may be already doing the check for us.

francoisferrand replied:
not sure we have this limitations...
test seems to pass, so we should be fine... unless we silently ignore the s3 error, as there is no assertion after `completeMpuObject` :-/

williamlardier replied:
We do have this limitation in cloudserver when doing a PutPart, but it does not consider the last part. This is only checked during CompleteMPU:

```

    // Note: Part sizes cannot be less than 5MB in size except for the last.
    // However, we do not check this value here because we cannot know which
    // part will be the last until a complete MPU request is made. Thus, we let
    // the completeMultipartUpload API check that all parts except the last are
    // at least 5MB.
```

bert-e commented:
# Request integration branches



Waiting for integration branch creation to be requested by the user.

To request integration branches, please comment on this pull request with the following command:

```
/create_integration_branches
```

Alternatively, the `/approve` and `/create_pull_requests` commands will automatically
create the integration branches.

francoisferrand committed:
ctst: check result of object creation steps

Issue: ZENKO-4928

francoisferrand committed:
Bump zenko-operator 1.6.5

Issue: ZENKO-4919

francoisferrand committed:
Bump cloudserver 8.8.36

Issue: ZENKO-4919

francoisferrand committed:
Bump sorbet 1.1.13

Issue: ZENKO-4919

francoisferrand committed:
Merge remote-tracking branch 'origin/bugfix/ZENKO-4925' into improvement/ZENKO-4919

francoisferrand committed:
Merge remote-tracking branch 'origin/improvement/ZENKO-4876' into improvement/ZENKO-4919

francoisferrand committed:
Release Zenko 2.6.70

Issue: ZENKO-4919

francoisferrand committed:
Release Zenko 2.7.66

Issue: ZENKO-4919

francoisferrand committed:
Fix flaky cluster issuer setup

We need to retry `kubectl apply`, esp as  mutating web hooks may not be
ready, c.f. [1].

[1] https://github.com/cert-manager/cert-manager/issues/2908

Issue: ZENKO-4919

francoisferrand committed:
Merge branch 'improvement/ZENKO-4919' into w/2.7/improvement/ZENKO-4919

francoisferrand commented:
/approve

bert-e commented:
# Incorrect fix version



The `Fix Version/s` in issue ZENKO-4928 contains:


* `2.10.7`

* `2.7.66`

* `2.8.46`

* `2.9.24`



Considering where you are trying to merge, I ignored possible hotfix versions and I expected to find:



* `2.10.8`

* `2.7.67`

* `2.8.47`

* `2.9.25`


Please check the `Fix Version/s` of ZENKO-4928, or the target
branch of this pull request.




*The following options are set:* **approve**

bert-e commented:
# 


I have successfully merged the changeset of this pull request
into targetted development branches:


* :heavy_check_mark: `development/2.7`

* :heavy_check_mark: `development/2.8`

* :heavy_check_mark: `development/2.9`

* :heavy_check_mark: `development/2.10`



The following branches have **NOT** changed:

* `development/2.5`
* `development/2.6`



Please check the status of the associated issue ZENKO-4928.

Goodbye francoisferrand.




*The following options are set:* **approve**
